docs/design/Abstract Syntax Tree.md

```diff
@@ -1,77 +1,33 @@
-is no different. A better term might have been "Abstract Semantic Tree" because our AST was defined with the goal of
-modeling the intent of the PartiQL source and *not* the exact syntax. Thus, the original SQL from which an AST was
-constituted cannot be derived, however the *semantics* of that SQL are guaranteed to be preserved. One example that
-demonstrates this is the fact that we model a `CROSS JOIN` in the same way that we model an `INNER JOIN` with a
-condition of `TRUE`. Semantically, these have the exact same function and so they also have the same representation in
-the AST.
+### Abstract Semantic Tree
+It seems the term "AST" is often extended to mean more things than just "Abstract Syntax Tree" and our use of the term is no different. A better term might have been "Abstract Semantic Tree" because our AST was defined with the goal of modeling the intent of the PartiQL source and *not* the exact syntax. Thus, the original SQL from which an AST was constituted cannot be derived, however the *semantics* of that SQL are guaranteed to be preserved. One example that demonstrates this is the fact that we model a `CROSS JOIN` in the same way that we model an `INNER JOIN` with a condition of `TRUE`. Semantically, these have the exact same functionality, so they have the same representation in the AST.
 
+### Tree Structure
 > It **is** Actually a Tree
 
 Language implementations often use the term "Abstract Syntax Tree" to refer to a data structure that is actually a
 graph. Our implementation for PartiQL's AST is a tree and *not* a graph. It contains no cycles and each node can only
 reference its children.
 
-## It's Immutable
+## Implementation
+There were multiple alternative implementations of PartiQL AST, e.g. S-expression & `ExprNode`, which were already deprecated & removed. Now we are using PIG generated AST for better maintainability. See https://github.com/partiql/partiql-ir-generator to learn more about PIG. The domain file is here, [partiql.ion](https://github.com/partiql/partiql-lang-kotlin/blob/main/lang/resources/org/partiql/type-domains/partiql.ion), where the tree structure just mirrors the PartiQL grammar. 
 
-No mechanism has been provided to mutate an instance of the AST after it has been instantiated.  Modifications to an
-existing tree must use cloning re-writes--for instance, a visitor pattern that returns a modified version of the input
-tree can be utilized.
+The rest of this doc describes some points of the implementation of PIG-generated code. 
 
-## Design Patterns Used with the AST
+### Immutable 
 
-The AST employs a number of patterns and utilizes certain features of Kotlin to aid with the development process.
-Without any introduction, the reasoning behind these patterns may not be completely apparent at first glance. What
-follows is an is an attempt to document those patterns and features.
-
-### Inheritance Mirrors The PartiQL Grammar
```

Comment:
I feel we don't need to explicitly mention how different kotlin classes map to AST nodes. Users can refer to `partiql.ion` file to understand it quickly.